Source/WebCore/Modules/compression/CompressionStreamEncoder.cpp

```diff
@@ -133,7 +132,7 @@ ExceptionOr<RefPtr<JSC::ArrayBuffer>> CompressionStreamEncoder::compress(const u
         if (result != Z_OK && result != Z_STREAM_END && result != Z_BUF_ERROR)
             return Exception { TypeError, "Failed to compress data."_s };
 
-        if (!m_zstream.avail_in) {
+        if (!m_zstream.avail_in && (!m_finish || (m_finish && result == Z_STREAM_END))) {
```

Comment:
`m_finish` is badly named. Boolean variables in WebKit need a prefix as per coding style. Should probably be something like `m_didFinish` or `m_isFinished`.

Reply:
`m_finish` is existing code, so renaming would increase the size of this bug fix. But it certainly seems like a good change to make.

I also feel like this conditional is confusing to read/understand. I think it's saying that if we have consumed all the input data, and are either not finished, or finished but didn't have enough output buffer space, we should expand the output buffer and continue.

I suggest you make a helper function with a name that explains this, and call it here (and in the similar DecompressionStream code) because the Zlib API is very terse and not easy to follow.